src/pyhf/infer/__init__.py

```diff
@@ -109,30 +119,38 @@ def hypotest(
     )
 
     _returns = [CLs]
+    if use_q0:
+        _returns = [CLsb]
+
     if kwargs.get('return_tail_probs'):
         _returns.append([CLsb, CLb])
-    if kwargs.get('return_expected_set'):
+    if kwargs.get('return_expected_set') or kwargs.get('return_expected'):
+        n_sigma_list = [2, 1, 0, -1, -2]
+        if not kwargs.get('return_expected_set'):
+            n_sigma_list = [0]
```

Comment:
added this to avoid having to type the code for calculating p0 and CLs (and doing the if-else) a second time for the case that only one expected value should be returned